app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt

```diff
@@ -49,22 +50,23 @@ class SearchUserFragment : Fragment(R.layout.fragment_search_user), OnItemClickL
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        vmSearchUser.view = view
+        vmSearchUser.ctx = view.context
 
-        setUpRecycleView(view)
-
-        searchEditText = view.findViewById(R.id.searchUserET)
+        setUpRecycleView()
+        searchEditText = vmSearchUser.view.findViewById(R.id.searchUserET)
         searchForUsers("")
 
-        setScanBtnListener(view)
+        setScanBtnListener()
 
         setSearchFieldListener()
     }
 
-    private fun setScanBtnListener(view: View) {
-        val buttonScan: FloatingActionButton = view.findViewById(R.id.searchUser_scanning)
+    private fun setScanBtnListener() {
+        val buttonScan: FloatingActionButton = vmSearchUser.view.findViewById(R.id.searchUser_scanning)
 
         buttonScan.setOnClickListener {
-            val qrIntent = Intent(requireActivity(), QrScanningActivity::class.java)
+            val qrIntent = Intent(activity, QrScanningActivity::class.java)
             qrIntent.putExtra("uidList", uidList)
             startActivity(qrIntent)
```

Comment:
I know you didn't wrote that part (only modify it)
But If you have time can you refactor it with the intent switcher.
Also if doable test it.